plugins/outputs/sql/sql.go

```diff
@@ -333,8 +339,107 @@ func (p *SQL) updateTableCache(tablename string) error {
 	return nil
 }
 
+func (p *SQL) processMetric(metric telegraf.Metric) (uint64, []string, []interface{}) {
+	// We need some hashing to later find the SQL statement belonging to
+	// the data to save regenerating the insert statement again. However,
+	// as we need to go through the metric already we should also do the
+	// hashing here to prevent reiterating all the fields and tags
+	var h maphash.Hash
+	h.Reset()
+	h.WriteString(metric.Name())
```

Comment:
Please end each write with a `null` or `newline` character as otherwise a metric with `name=foo`, `time-column=bar` and tags `a`, `b` will create the same hash as `name=foo`, `time-column=bar` and tag `ab`...

```suggestion
	h.WriteString(metric.Name()+"\n")
```

same for all other string writes...